core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+    ts_parser.add_argument("--nperm", metavar="n", type=int, default=1000,
+                           help="number of permutations for p-value estimation (default: 1000)")
+    
+    # This parser is unnecessary for now, but provided for easier algorithm implementation
+    sp_parser = subparsers.add_parser("SigProfiler")
+
+    args = parser.parse_args()
+    return args
+
+
+def load_yaml(file):
+    """Simple load a yaml file. Only accepts key: value pairs, no multi-line statements
+    
+    :param file: object, open yaml file 
+    :return: dict, yaml key-value pairs
+    """
+    yaml_cont = {}
+    for line in file:
+        split_line = line.strip().split(":")
+        key = split_line[0].strip()
+        val = "".join(split_line[1:]).strip()
+        yaml_cont[key] = val
+    return yaml_cont
+
+
+def find_mutations_data(study_path):
+    """Find mutation data (MAF) file in cBioPortal study folder
+    
+    :param study_path: str, path to cBioPortal study folder
+    :return: str, name of mutations data (MAF) file
+    """
+    meta_files = glob.glob(study_path + "/meta*")
+    if len(meta_files) == 0:
+        raise FileNotFoundError("No meta files present in study")
+    for file in meta_files:
+        with open(file) as f:
+            yaml_cont = load_yaml(f)
+        if yaml_cont.get("stable_id") == "mutations" and yaml_cont.get("datatype") == "MAF":
+            # Correct file, retrieve associated datafile
+            print(f"Mutation meta file found in study: \'{file}\'")
+            return yaml_cont.get("data_filename"), yaml_cont.get("cancer_study_identifier")
+    raise FileNotFoundError(f"Mutation meta file not found in study \'{study_path}\'.")
+
+
+def acquire_maf_study_id(study_path):
+    """Find meta file in study and retrieve mutations file and study ID
+    
+    :param study_path: str, path to cBioPortal study
+    :return: str, mutation file path; str, study ID    
+    """
+    # Check if study exist
+    if not os.path.exists(study_path):
+        raise FileNotFoundError(f"{study_path} does not exist")
+    # Find meta files in study
+    mut_file, study_id = find_mutations_data(study_path)
+    mut_file = f"{study_path}/{mut_file}"
+    return mut_file, study_id
+
+
+def decipher_genome(maf_df):
+    """Try to get the genome build from a MAF file
+    
+    :param maf_df: pandas object, pandas dataframe containing 'NCBI_Build' column
+    :return: str, genome build (GRCh**)
+    """
+    genome = list(maf_df["NCBI_Build"].unique())
+    if len(genome) != 1:
+        raise ValueError(f"There are either no or multiple genome builds present in your MAF file: \'{genome}\'")
+    try:
+        genome = int(genome[0])
+    except ValueError:
+        try:
+            genome = int(genome[0][-2:])
+        except ValueError:
+            genome = input("Can't decipher genome build. Please enter manually (37/38): ")
+            genome = int(genome)
+    genome = "GRCh" + str(genome)
```

Comment:
Following the comment from Sander, could this be made with a mapping of accepted possibilities?
`{'19': 'GRCh37', 'hg19': 'GRCh37', ...}`
And finally
`if not ncbi_build.lower() in ['grch37', 'grch38']: ERROR`
The current code would accept e.g. GRCh25

Reply:
I took the name `ncbi_build` from `validateData.py`, and maybe it's good to keep it consistent?

Reply:
Nice, good catch! I will make a mapper! Do you also suggest removing the input option and adding a genome cml parameter? as to keep the program without user input?